source/m2mconnectionhandlerpimpl.cpp

```diff
@@ -434,6 +435,7 @@ bool M2MConnectionHandlerPimpl::resolve_hostname(const char *address,
     if(_socket_server > 0) {
         close(_socket_server);
         _socket_server = -1;
+        pthread_join(_listen_thread, NULL);
```

Comment:
Document says following:
"If that thread has already terminated, then  pthread_join() returns immediately."
But there is also statement:
"Joining with a thread that has previously been joined results in  undefined behavior."

So it's pretty confusing... Valgrind didn't spot any issues when pthread_join was executed multiple times.